tests/std/tests/LWG2510_tag_classes/test.compile.pass.cpp

```diff
@@ -0,0 +1,207 @@
+
+STATIC_ASSERT(implicitly_default_constructible<int>);
+STATIC_ASSERT(implicitly_default_constructible<void*>);
+
+STATIC_ASSERT(!implicitly_default_constructible<void>);
+STATIC_ASSERT(!implicitly_default_constructible<int&>);
+STATIC_ASSERT(!implicitly_default_constructible<void()>);
+
+template <class T>
+void check_tag_class() { // COMPILE-ONLY
+    STATIC_ASSERT(is_default_constructible_v<T>);
+    STATIC_ASSERT(!implicitly_default_constructible<T>);
+
+    STATIC_ASSERT(is_copy_constructible_v<T>);
+    STATIC_ASSERT(is_move_constructible_v<T>);
+    STATIC_ASSERT(is_constructible_v<T, T&>);
+    STATIC_ASSERT(is_constructible_v<T, const T>);
+    STATIC_ASSERT(is_convertible_v<T&, T>);
+    STATIC_ASSERT(is_convertible_v<const T&, T>);
+    STATIC_ASSERT(is_convertible_v<T, T>);
+    STATIC_ASSERT(is_convertible_v<const T, T>);
+    STATIC_ASSERT(is_copy_assignable_v<T>);
+    STATIC_ASSERT(is_move_assignable_v<T>);
+    STATIC_ASSERT(is_assignable_v<T&, T&>);
+    STATIC_ASSERT(is_assignable_v<T&, const T>);
+
+    STATIC_ASSERT(is_nothrow_destructible_v<T>);
+
+    STATIC_ASSERT(!is_polymorphic_v<T>);
+
+    constexpr T constant_argument;
+    (void) constant_argument;
+
+    // desired properties not yet guaranteed by the Standard
+    STATIC_ASSERT(is_nothrow_default_constructible_v<T>);
+    STATIC_ASSERT(is_nothrow_copy_constructible_v<T>);
+    STATIC_ASSERT(is_nothrow_move_constructible_v<T>);
+    STATIC_ASSERT(is_nothrow_constructible_v<T, T&>);
+    STATIC_ASSERT(is_nothrow_constructible_v<T, const T>);
+#if _HAS_CXX20
+    STATIC_ASSERT(is_nothrow_convertible_v<T&, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<const T&, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<T, T>);
+    STATIC_ASSERT(is_nothrow_convertible_v<const T, T>);
+#endif // _HAS_CXX20
+    STATIC_ASSERT(is_nothrow_copy_assignable_v<T>);
+    STATIC_ASSERT(is_nothrow_move_assignable_v<T>);
+    STATIC_ASSERT(is_nothrow_assignable_v<T&, T&>);
+    STATIC_ASSERT(is_nothrow_assignable_v<T&, const T>);
+
+    STATIC_ASSERT(is_trivially_default_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_copy_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_move_constructible_v<T>);
+    STATIC_ASSERT(is_trivially_constructible_v<T, T&>);
+    STATIC_ASSERT(is_trivially_constructible_v<T, const T>);
+    STATIC_ASSERT(is_trivially_copy_assignable_v<T>);
+    STATIC_ASSERT(is_trivially_move_assignable_v<T>);
+    STATIC_ASSERT(is_trivially_assignable_v<T&, T&>);
+    STATIC_ASSERT(is_trivially_assignable_v<T&, const T>);
+    STATIC_ASSERT(is_trivially_destructible_v<T>);
+    STATIC_ASSERT(is_trivially_copyable_v<T>);
+
+    STATIC_ASSERT(is_standard_layout_v<T>);
+    STATIC_ASSERT(is_empty_v<T>); // only guaranteed for allocator_arg_t and piecewise_construct_t
+
+    // ABI-specific, seemingly agreed among mainstream implementations
+    STATIC_ASSERT(sizeof(T) == 1);
+}
+
+void check_tag_classes() { // COMPILE-ONLY
+    check_tag_class<nothrow_t>();

```

Comment:
Naming nitpick: It doesn't really matter because this is just a compile-only bundle of tests, but `check_tag_classes()` both looks a lot like the singular `check_tag_class()`, and yet isn't specific enough. This function is checking the *Standard* tags, as opposed to the implementation details below. `check_standard_tags()` would make this distinction clearer.